Property collector throws a runtime exception when retrieving VM update

Sometimes, when the blocking method **waitForUpdatesEx** is released, it throws the following exception:

```
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
    java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
    at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.eaton.pqsoft.virtualization.vmware.VMwarePropertiesMonitor$Poller.run(VMwarePropertiesMonitor.java:372)[136:virtualization-vmware-impl:0.2.1]
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_40]
    at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)[:1.8.0_40]
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)[:1.8.0_40]
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)[:1.8.0_40]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_40]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_40]
    at java.lang.Thread.run(Thread.java:745)[:1.8.0_40]
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
    at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)[134:com.eaton.pqsoft.emc4j.thirdparty.yavijava-bundle:6.0.4]
    ... 12 more
```

It appears when the vcenter sents an update of a VM, typically a power state change.
We use the version 6.0.04 of Yavijava with Java 8.
Thanks.
Christian

Here you go, @michaelrice:

/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/bin/java -agentlib:jdwp=transport=dt_socket,address=127.0.0.1:60457,suspend=y,server=n -Dfile.encoding=UTF-8 -classpath "/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/charsets.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/deploy.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/ext/dnsns.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/ext/localedata.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/ext/sunec.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/ext/sunjce_provider.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/ext/sunpkcs11.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/ext/zipfs.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/htmlconverter.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/javaws.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/jce.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/jfr.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/jfxrt.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/jsse.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/management-agent.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/plugin.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/resources.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/lib/rt.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/lib/ant-javafx.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/lib/dt.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/lib/javafx-doclet.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/lib/javafx-mx.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/lib/jconsole.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/lib/sa-jdi.jar:/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/lib/tools.jar:/Users/solidfire/Documents/yavijava/YAVIJAVA-Test/target/classes:/Users/solidfire/.m2/repository/com/toastcoders/yavijava/6.0.04/yavijava-6.0.04.jar:/Users/solidfire/.m2/repository/org/apache/httpcomponents/httpclient/4.3.5/httpclient-4.3.5.jar:/Users/solidfire/.m2/repository/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar:/Users/solidfire/.m2/repository/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar:/Users/solidfire/.m2/repository/commons-codec/commons-codec/1.6/commons-codec-1.6.jar:/Users/solidfire/.m2/repository/org/apache/directory/studio/org.dom4j.dom4j/1.6.1/org.dom4j.dom4j-1.6.1.jar:/Users/solidfire/.m2/repository/dom4j/dom4j/1.6.1/dom4j-1.6.1.jar:/Users/solidfire/.m2/repository/log4j/log4j/1.2.17/log4j-1.2.17.jar:/Users/solidfire/.m2/repository/org/projectlombok/lombok/1.16.6/lombok-1.16.6.jar:/Users/solidfire/.m2/repository/org/apache/logging/log4j/log4j-api/2.8.2/log4j-api-2.8.2.jar:/Users/solidfire/.m2/repository/org/apache/logging/log4j/log4j-core/2.8.2/log4j-core-2.8.2.jar:/Applications/IntelliJ IDEA CE.app/Contents/lib/idea_rt.jar" Yavijava_Test
Connected to the target VM, address: '127.0.0.1:60457', transport: 'socket'
0 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. ServiceContent
235 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. ServiceContent
439 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. UserSession
607 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. ManagedObjectReference
734 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. ManagedObjectReference
981 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. UpdateSet
1
37590 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. UpdateSet
37616 [main] ERROR com.vmware.vim25.ws.WSClient  - Exception caught while invoking method: WaitForUpdatesEx
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
We have an error.
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
43487 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. UpdateSet
We have an error.
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
43491 [main] ERROR com.vmware.vim25.ws.WSClient  - Exception caught while invoking method: WaitForUpdatesEx
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
43644 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. UpdateSet
43647 [main] ERROR com.vmware.vim25.ws.WSClient  - Exception caught while invoking method: WaitForUpdatesEx
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
We have an error.
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
43799 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. UpdateSet
We have an error.
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
43809 [main] ERROR com.vmware.vim25.ws.WSClient  - Exception caught while invoking method: WaitForUpdatesEx
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more
43940 [main] DEBUG com.vmware.vim25.ws.XmlGenDom  - Parsing XML payload from server. UpdateSet
We have an error.
java.rmi.RemoteException: Exception in SoapClient.invoke:; nested exception is: 
	java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:125)
	at com.vmware.vim25.ws.SoapClient.unMarshall(SoapClient.java:253)
	at com.vmware.vim25.ws.WSClient.invoke(WSClient.java:96)
	at com.vmware.vim25.ws.VimStub.waitForUpdatesEx(VimStub.java:134)
	at com.vmware.vim25.mo.PropertyCollector.waitForUpdatesEx(PropertyCollector.java:119)
	at Yavijava_Test.main(Yavijava_Test.java:53)
Caused by: java.lang.RuntimeException: Unexpected Type at setObjectField: java.lang.Objectval
	at org.doublecloud.ws.util.ReflectUtil.setObjectField(ReflectUtil.java:130)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:332)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXml(XmlGenDom.java:349)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:237)
	at com.vmware.vim25.ws.XmlGenDom.fromXML(XmlGenDom.java:122)
	... 5 more

--- a/src/main/java/org/doublecloud/ws/util/ReflectUtil.java
+++ b/src/main/java/org/doublecloud/ws/util/ReflectUtil.java
@@ -126,6 +126,9 @@ public class ReflectUtil {
         else if ("Double".equals(type)) {
             field.set(object, new Double(value));
         }
+        else if ("base64Binary".equals(type)) {
+            field.set(object, DatatypeConverter.parseBase64Binary(value));
+        }
         else {
             throw new RuntimeException("Unexpected Type at setObjectField: " + field.getType().getCanonicalName() + field.getName());
         }

--- a/src/test/java/com/vmware/vim25/ws/XmlGenDomTest.java
+++ b/src/test/java/com/vmware/vim25/ws/XmlGenDomTest.java
@@ -155,4 +155,11 @@ public class XmlGenDomTest {
         String[] strings = (String[]) xmlGenDom.fromXML("String[]", inputStream);
         assert strings.getClass().isArray();
     }
+
+    @Test
+    public void testFromXML_When_UpdateSet_Contains_Base64_Binary_No_Exception_Is_Thrown() throws Exception {
+        InputStream inputStream = new FileInputStream(new File("src/test/resources/xml/UpdateSetWithBase64Binary.xml"));
+        XmlGenDom xmlGenDom = new XmlGenDom();
+        UpdateSet updateSet = (UpdateSet) xmlGenDom.fromXML("UpdateSet", inputStream);
+    }
 }

--- a/src/test/java/org/doublecloud/ws/util/ReflectUtilTest.java
+++ b/src/test/java/org/doublecloud/ws/util/ReflectUtilTest.java
@@ -1,6 +1,7 @@
 package org.doublecloud.ws.util;
 
 import com.vmware.vim25.AboutInfo;
+import com.vmware.vim25.PropertyChange;
 import org.doublecloud.ws.util.testUtils.*;
 import org.junit.Assert;
 import org.junit.Test;
@@ -61,5 +62,16 @@ public class ReflectUtilTest {
         strings.add("string3");
         String[] stringArray = (String[]) ReflectUtil.parseToObject(type, strings);
         assert stringArray.getClass().isArray();
+    }
+
+    @Test
+    public void testReflectUtil_SetObjectField_Supports_Base64Binary_To_ByteArray() throws Exception {
+        String base64BinaryStr = "EtUGWJdr2BYg3Dom7G6oPAlHHcc=";
+        Object obj = new PropertyChange();
+        Field field = PropertyChange.class.getField("val");
+        String type = "base64Binary";
+        ReflectUtil.setObjectField(obj, field, type, base64BinaryStr);
+        PropertyChange pc = (PropertyChange) obj;
+        assert pc.getVal() instanceof byte[];
     }
 }

--- a/src/test/resources/xml/UpdateSetWithBase64Binary.xml
+++ b/src/test/resources/xml/UpdateSetWithBase64Binary.xml
@@ -1,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <soapenv:Body>
+        <WaitForUpdatesExResponse xmlns="urn:vim25">
+            <returnval>
+                <version>3</version>
+                <filterSet>
+                    <filter type="PropertyFilter">session[5293e761-ee4a-d9cd-1dc3-8fe9c6ef7939]52379183-1a11-af2a-0e0a-c8ce4c507066</filter>
+                    <objectSet>
+                        <kind>modify</kind>
+                        <obj type="VirtualMachine">vm-13</obj>
+                        <changeSet>
+                            <name>config.annotation</name>
+                            <op>assign</op>
+                            <val xsi:type="xsd:string"/>
+                        </changeSet>
+                        <changeSet>
+                            <name>config.changeVersion</name>
+                            <op>assign</op>
+                            <val xsi:type="xsd:string">2017-04-30T18:10:33.370382Z</val>
+                        </changeSet>
+                        <changeSet>
+                            <name>config.vmxConfigChecksum</name>
+                            <op>assign</op>
+                            <val xsi:type="xsd:base64Binary">EtUGWJdr2BYg3Dom7G6oPAlHHcc=</val>
+                        </changeSet>
+                    </objectSet>
+                </filterSet>
+            </returnval>
+        </WaitForUpdatesExResponse>
+    </soapenv:Body>
+</soapenv:Envelope>

